### ISSUE TITLE
implicit subscription not working on heterogeneous silos

### ISSUE THREAD

**Issue by mkjeff** [dotnet/orleans#5557]
I have a heterogeneous silos environment. 
Silo1: GrainA class with an implicit subscription on the namespace 'topic'
Silo2: without GrainA

When the associated queue's pull-agent was activated on Silo2, the pull-agent can't obtain the GrainA information to complete `implicitPubSub.RegisterConsumer` and seems `ImplicitStreamSubscriberTable` was not transferred between silos. 

I'm also wondering that is it possible to make `ImplicitStreamSubscriptionAttribute` be able to apply to the grain interface?

**Comment by benjaminpetit** [comment]
You are right, it's a limitation in heretogeneous silos. It will need some work to enable this scenario, but I don't know enough the streaming infrastructure to have an estimate of the amount of work needed.

@jason-bragg ?

**Comment by jason-bragg** [comment]
At a glance, it looks like it should work if we simply allow the attribution to be added to class or interface.  We'd need to test that though.

@mkjeff, Adding request to backlog.  Fastest path to addressing this would be to change the attribute in your own fork, see if it works, and if so submit a PR.

**Comment by mkjeff** [comment]
@jason-bragg,  I will try a bit, thanks for your advice.

**Comment by Rikarin** [comment]
Has there been any progress on this?

**Comment by Rikarin** [comment]
@ReubenBond I haven't experienced the issue and I wasn't sure if it's an edge case or it was already fixed as it's still mentioned in the docs https://learn.microsoft.com/en-us/dotnet/orleans/host/heterogeneous-silos
